### DISCUSSION TITLE
What is the encryption algorithm of the user password？

### DISCUSSION THREAD

**Discussion by cykzht** [Pryaxis/TShock#2489] [Q&A]
I want to use PHP to develop a server console. In the login part of the web page, I want to use the password stored in the database by the server.
In config.json I set
`"HashAlgorithm": "sha512"`
But I use sha512 algorithm in PHP to process the password and it cannot match the content in the database.
So I want to ask what is the specific process of this encryption, and how I can implement such an encryption algorithm in PHP.
![img](https://www.g2022cyk.top:85/wp-content/uploads/2021/11/1636860158-data.png)

**Comment by hakusaro** [comment]
It's [bcrypt](https://en.wikipedia.org/wiki/Bcrypt). You might find this helpful: https://stackoverflow.com/questions/4795385/how-do-you-use-bcrypt-for-hashing-passwords-in-php
